### ISSUE TITLE
@Poly* in type argument position

### ISSUE THREAD

**Issue by wmdietlGC** [typetools/checker-framework#2432]
Take the following Nullness Checker example that uses `@PolyNull` in a type argument:

```java
import org.checkerframework.checker.nullness.qual.Nullable;
import org.checkerframework.checker.nullness.qual.PolyNull;
import java.util.List;

class PolyNullTest {
  void foo(List<@PolyNull Object> l) {
    l.add(null);
  }
  void test(List<@Nullable Object> n, List<Object> nn) {
      foo(n);
      foo(nn);
  }
}
```

This correctly allows both calls of `foo`, but does not raise an error for `l.add(null)`.
As this could be executed on the non-null list, adding `null` must be forbidden as it could lead to an NPE.

If the signature of `foo` is changed to `List<? extends @Nullable Object>` the addition of `null` is correctly prevented.
Using `@PolyNull` for this is useful when using a wildcard isn't meaningful, e.g. for a bound that is a final class.

**Comment by smillst** [comment]
This is a general problem.  The follow code should issue a warning, but doesn't:
```
import java.util.List;
import org.checkerframework.checker.tainting.qual.PolyTainted;
import org.checkerframework.checker.tainting.qual.Tainted;

class PolyTaintedTest {
    void polyTainted(List<@PolyTainted Object> list, @Tainted String s) {
        list.add(s);
    }
}
```
